### PR TITLE
update client info parsing to accept IPv6 addr

### PR DESCRIPTION
**Pull request by edward-milkey**
This PR addresses issue #863. `parseClientListString` is updated to parse IPv6 addresses.

### CODE REVIEW

File: exporter/clients.go [hunk 1]
```diff
@@ -71,11 +72,12 @@ func parseClientListString(clientInfo string) (*ClientInfo, bool) {
...
 		case "addr":
 			hostPortString := strings.Split(vPart[1], ":")
-			if len(hostPortString) != 2 {
+			if len(hostPortString) < 2 {
+				log.Debug("Invalid value for 'addr' found in client info")
 				return nil, false
 			}
-			connectedClient.Host = hostPortString[0]
-			connectedClient.Port = hostPortString[1]
+			connectedClient.Host = strings.Join(hostPortString[:len(hostPortString)-1], ":")
```
Comment (oliver006) [inline exporter/clients.go:79]: How's that working with IPv4? In that case we just have one item in the "Join()" and no ":" will be added?

File: exporter/clients.go [hunk 2]
```diff
@@ -29,9 +29,10 @@ type ClientInfo struct {
 	Valid Examples
 	id=11 addr=127.0.0.1:63508 fd=8 name= age=6321 idle=6320 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=0 qbuf-free=0 obl=0 oll=0 omem=0 events=r cmd=setex user=default resp=2
 	id=14 addr=127.0.0.1:64958 fd=9 name= age=5 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=26 qbuf-free=32742 obl=0 oll=0 omem=0 events=r cmd=client user=default resp=3
+	id=40253233 addr=fd40:1481:21:dbe0:7021:300:a03:1a06:44426 fd=19 name= age=782 idle=0 flags=N db=0 sub=0 psub=0 multi=-1 qbuf=26 qbuf-free=32742 argv-mem=10 obl=0 oll=0 omem=0 tot-mem=61466 ow=0 owmem=0 events=r cmd=client user=default lib-name=redis-py lib-ver=5.0.1 numops=9
 */
 func parseClientListString(clientInfo string) (*ClientInfo, bool) {
-	if matched, _ := regexp.MatchString(`^id=\d+ addr=\d+`, clientInfo); !matched {
+	if matched, _ := regexp.MatchString(`^id=\d+ addr=\w+`, clientInfo); !matched {
```
Comment (oliver006) [inline exporter/clients.go:35]: nit: is it worth changing this from \d or \w to "not a (white)space character"?

**Review by oliver006 — APPROVED** [review]
Thanks for the PR!
Two minor comments/questions but this looks good.

File: exporter/clients.go (hunk 2 above)
Comment (edward-milkey) [reply exporter/clients.go]: Yes, I like your suggestion better. Updated.

File: exporter/clients.go (hunk 1 above)
Comment (edward-milkey) [reply exporter/clients.go:79]: Hi @oliver006 - thanks for the quick review!

Yes, that's correct. When there's 1 item in `Join` the join character is not added.

File: exporter/clients.go (hunk 2 above)
Comment (oliver006) [reply exporter/clients.go]: Looks good, thanks for updating!

**Pull request merged**